### ISSUE TITLE
Bug when interpolating multiple fields for output at points

### ISSUE THREAD

**Issue by baagaard-usgs** [geodynamics/pylith#320]
@knepley There is a bug in `DMInterpolationEvaluate()` when interpolating for multiple fields. The `xa` array holds all of the fields, not just the current field in the loop. The index into the `xa` array needs to be offset so that it works for more than one field.

```
for (f = 0; f < Nb; ++f) {
  interpolant[p*ctx->dof+c+fc] += xa[f]*basis[(0*Nb + f)*Nc + fc];
//                                  ^^^ This is wrong. It needs to be off+f, where off is the offset for this field.
}
```

**Comment by knepley** [comment]
@baagaard-usgs I just pushed  fix to pylith-dev. Let me know if it does not work.

**Comment by baagaard-usgs** [comment]
Fixed in PETSc 8182922e30a57d4449b615f5324c3b50fa4ea660
